python/lsst/consdb/utils.py

```diff
@@ -0,0 +1,100 @@
+# This file is part of consdb.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Utility functions for consdb services.
+"""
+import logging
+import os
+import re
+import sys
+
+import sqlalchemy
+
+__all__ = ["setup_postgres", "setup_logging"]
+
+
+def setup_postgres() -> sqlalchemy.Engine:
+    """Set up a SQLAlchemy Engine to talk to Postgres.
+
+    Uses environment variables to get connection information, with an internal
+    default.
+
+    Returns
+    -------
+    engine: ``sqlalchemy.Engine``
+        A SQLAlchemy Engine.
+    """
+    host = os.environ.get("DB_HOST")
+    passwd = os.environ.get("DB_PASS")
+    user = os.environ.get("DB_USER")
+    dbname = os.environ.get("DB_NAME")
+    pg_url = ""
+    if host and passwd and user and dbname:
+        logging.info(f"Connecting to {host} as {user} to {dbname}")
+        pg_url = f"postgresql://{user}:{passwd}@{host}/{dbname}"
+    else:
+        pg_url = os.environ.get(
+            "POSTGRES_URL", "postgresql://usdf-butler.slac.stanford.edu:5432/lsstdb1"
+        )
+        logging.info(f"Using POSTGRES_URL {user} {host} {dbname}")
+    return sqlalchemy.create_engine(pg_url)
+
+
+def setup_logging(module: str) -> logging.Logger:
```

Comment:
Perhaps we should consider using Safir ?  That would give a standardize configuration and logging (e..g above for DB_HOST could be config.db_host instead of explicitly from the env.   But for now perhaps we leave it.

Reply:
and logging of course

Reply:
Yeah, the whole thing could be rewritten with Safir, but it's meant to be a short-term shim until Sasquatch and TAP can be used instead.